### DISCUSSION TITLE
zTunnel is not getting up when run as non-root user

### DISCUSSION THREAD

**Discussion by smoharana1977** [istio/istio#53657] [Q&A] [answered]
hello, when I have updated securityContext for ztunnel daemonset to userconext --> nonrootuser: true, the ztunnel pods are not getting up and throwing below error - 

2024-10-25T18:05:15.927384Z	info	ztunnel	version: version.BuildInfo{Version:"245ecc111ba1cbe9bd2923f24be609e656a191ad", GitRevision:"245ecc111ba1cbe9bd2923f24be609e656a191ad", RustVersion:"1.79.0", BuildProfile:"release", BuildStatus:"Clean", GitTag:"1.23.0-3-g245ecc1", IstioVersion:"unknown"}	

2024-10-25T18:05:15.928082Z	info	app	shared proxy mode - in-pod mode enabled	
2024-10-25T18:05:15.929536Z	info	readiness	Task 'dns proxy' complete (2.666919ms), still awaiting 2 tasks	
2024-10-25T18:05:15.929546Z	info	readiness	Task 'proxy' complete (2.677596ms), still awaiting 1 tasks	
Error: failed to start workload proxy manager failed to set netns: Os { code: 1, kind: PermissionDenied, message: "Operation not permitted" }

**Comment by howardjohn** [comment]
The securityContext of ztunnel in the default installation has already be scrutinized to be as minimal as possible. It requires this

**Reply by smoharana1977** [reply]
we applied the policy in cluster level for not running the pods using rootuser and after that ztunnel daemonset was not starting. Hence updated the securitycontext --> nonrootuser = true

error when nonrootuser=false
  Error creating: admission webhook "validation.gatekeeper.sh" denied the
  request: [azurepolicy-xxxxx]
  Container istio-proxy is attempting to run as disallowed user 0. Allowed
  runAsUser: {"ranges": [], "rule": "MustRunAsNonRoot"}

**Reply by howardjohn** [reply]
You will need to make an allowance for this to run. Note that all networking proxies like this (kube-proxy, cilium, calico, etc) inherently require the same/similar permissions to function.

**Reply by smoharana1977** [reply]
but the issue with only zTunnel and ingress/egress/cni are running good with non root user policy. Are you saying we should relax the policy for istio-system namespaces?

**Reply by howardjohn** [reply]
I don't think CNI can run as non-root either. Ztunel and CNI both require root. Ingress, egress, and istiod do not.

**Reply by smoharana1977** [reply]
I checked and it seems CNI is running as root user. However the policy is not impacting it. Any other input to check?
NAMESPACE      NAME                                       DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR              AGE
istio-system   ztunnel                                    8         8         0       0            0           kubernetes.io/os=linux     23h
kube-system    istio-cni-node                             8         8         8       8            8           kubernetes.io/os=linux     23h

**Reply by smoharana1977** [reply]
based on policy, it relaxed for kube-system ns and moved ztunnel to there, it worked.

Do you see any issues for running ztunnel under kube-system?

**Reply by howardjohn** [reply, marked as answer]
No, it can run in any namespace
